### PR TITLE
lxc: 4.0.5 -> 4.0.6

### PR DESCRIPTION
**Pull request by marsam**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/lxc/default.nix [hunk 1]
```diff
@@ -41,10 +40,10 @@ stdenv.mkDerivation rec {
     "--disable-api-docs"
     "--with-init-script=none"
     "--with-distro=nixos" # just to be sure it is "unknown"
-  ] ++ optional (libapparmor != null) "--enable-apparmor"
-    ++ optional (libselinux != null) "--enable-selinux"
-    ++ optional (libseccomp != null) "--enable-seccomp"
-    ++ optional (libcap != null) "--enable-capabilities"
+  ] ++ lib.optional (libapparmor != null) "--enable-apparmor"
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/lxc/default.nix:43]: Shouldn't this be flags like withAppArmor?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109071` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
8 packages built:
</summary>
<ul>
<li>anbox</li>
<li>haskellPackages.bindings-lxc</li>
<li>lxc</li>
<li>lxd</li>
<li>pipework</li>
<li>rubyPackages.ruby-lxc (rubyPackages_2_6.ruby-lxc)</li>
<li>rubyPackages_2_5.ruby-lxc</li>
<li>rubyPackages_2_7.ruby-lxc</li>
</ul>
</details>

File: pkgs/os-specific/linux/lxc/default.nix (hunk 1 above)
Comment (marsam) [reply pkgs/os-specific/linux/lxc/default.nix:43]: yes, but I'm not sure how to change it without breaking other people setups

Comment (SuperSandro2000) [reply pkgs/os-specific/linux/lxc/default.nix:43]: We could default it to true.

Comment (marsam) [reply pkgs/os-specific/linux/lxc/default.nix]: yes, I think it should be fine, but I'd prefer changing it in separate PR

**Pull request merged**